pkg/cleaner/azure/resourcegroup.go

```diff
@@ -0,0 +1,97 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-02-01/resources"
+	"github.com/giantswarm/microerror"
+)
+
+const (
+	// gracePeriod represents the maximum time the CI resources are allowed to
+	// remain up. CI resources older than gracePeriod will be deleted.
+	gracePeriod = 90 * time.Minute
+)
+
+func (c Cleaner) cleanResourceGroup(ctx context.Context) error {
+	var lastError error
+
+	// It would be more efficient here to use a filter like "startswith(name,'ci-') or startswith(name,'e2e')"
+	// but this does not seems to work now, see https://github.com/Azure/azure-sdk-for-go/issues/2480.
+	groupIter, err := c.groupsClient.ListComplete(ctx, "", nil)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	deadLine := time.Now().Add(-gracePeriod).UTC()
+
+	for ; groupIter.NotDone(); groupIter.Next() {
+		group := groupIter.Value()
+
+		c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("check resource group %q", *group.Name))
+
+		shouldBeDeleted, err := c.groupShouldBeDeleted(ctx, group, deadLine)
+		if err != nil {
+			c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("failed to check resource group %q", *group.Name), "error", err.Error())
+			r.logger.LogCtx(ctx, "level", "debug", "message", "skipping")
+			lastError = err
+			continue
+		}
+
+		if shouldBeDeleted {
+			c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensuring deletion of resource group %q", *group.Name))
+
+			respFuture, err := c.groupsClient.Delete(ctx, *group.Name)
+			if err != nil {
+				c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("did not ensure deletion for resource group %q ", *group.Name), "error", err.Error())
```

Comment:
We don't have any known `error` key in log messages AFAIK. It would be good to wrap the error in `Mask` and print it in the `stack` key. That would be more standard.

Reply:
done